modules/lfs/content_store.go

```diff
@@ -82,6 +82,15 @@ func (s *ContentStore) Exists(meta *models.LFSMetaObject) bool {
 	return true
 }
 
+// Verify returns true if the object exists in the content store and size is correct.
+func (s *ContentStore) Verify(meta *models.LFSMetaObject) bool {
+	path := filepath.Join(s.BasePath, transformKey(meta.Oid))
+	if fi, err := os.Stat(path); os.IsNotExist(err) || fi.Size() != meta.Size {
```

Comment:
What if `err` is not `nil`, but is something other than a `IsNotExist` error. I think we should return `false` in that case.

Reply:
@erhankoenig I did not want to change that as that is current behaviour also for exists method

Reply:
But if `err` is not nil, it could be the case that `fi` is nil. In that case, `fi.Size()` will result in a nil-pointer dereference, so I think we need to check if `err` is nil or not.

Reply:
But if `err` is not nil, it could be the case that `fi` is nil. In that case, `fi.Size()` will result in a nil-pointer dereference, so I think we need to check if `err` is nil or not.

Reply:
That's true but I don't know if in such case we should return false